dali/operators/image/paste/multipaste.h

```diff
@@ -0,0 +1,217 @@
+    for (int i = 0; i < curr_batch_size; i++) {
+      const int64_t n_paste = in_idx_[i].shape[0];
+
+      if (in_anchors_.IsDefined()) {
+        DALI_ENFORCE(in_anchors_[i].shape[0] == n_paste,
+                     "in_anchors must be same length as in_idx");
+        DALI_ENFORCE(in_anchors_[i].shape[1] == spatial_ndim,
+         "in_anchors must have number of coordinates equal to that of input images - 1 (channel)");
+      }
+      if (in_shapes_.IsDefined()) {
+        DALI_ENFORCE(in_shapes_[i].shape[0] == n_paste, "in_shapes must be same length as in_idx");
+        DALI_ENFORCE(in_shapes_[i].shape[1] == spatial_ndim,
+           "in_shapes must have number of coordinates equal to that of input images - 1 (channel)");
+      }
+      if (out_anchors_.IsDefined()) {
+        DALI_ENFORCE(out_anchors_[i].shape[0] == n_paste,
+                     "out_anchors must be same length as in_idx");
+        DALI_ENFORCE(out_anchors_[i].shape[1] == spatial_ndim,
+         "out_anchors must have number of coordinates equal to that of input images - 1 (channel)");
+      }
+
+      bool found_intersection = false;
+
+      for (int j = 0; j < n_paste; j++) {
+        auto out_anchor = GetInAnchors(i, j);
+        auto j_idx = in_idx_[i].data[j];
+        const auto &shape = GetShape(i, j, Coords(raw_input_size_mem_.data() + 2 * j_idx,
+                                                  dali::TensorShape<>(2)));
+        for (int k = 0; k < j; k++) {
+          auto k_idx = in_idx_[i].data[k];
+          if (Intersects(out_anchor, shape, GetInAnchors(i, k), GetShape(
+                  i, k, Coords(raw_input_size_mem_.data() + 2 * k_idx, dali::TensorShape<>(2))))) {
+            found_intersection = true;
+            break;
+          }
+        }
+        if (found_intersection) {
+          break;
+        }
+      }
+      no_intersections_.push_back(!found_intersection);
+    }
+  }
+
+  inline Coords GetInAnchors(int sample_num, int paste_num) const {
+    return in_anchors_.IsDefined()
+           ? subtensor(in_anchors_[sample_num], paste_num)
+           : zero_anchors_;
+  }
+
+  inline Coords GetShape(int sample_num, int paste_num, Coords default_shape) const {
+    return in_shapes_.IsDefined()
+           ? subtensor(in_shapes_[sample_num], paste_num)
+           : default_shape;
+  }
+
+  inline Coords GetOutAnchors(int sample_num, int paste_num) const {
+    return out_anchors_.IsDefined()
+           ? subtensor(out_anchors_[sample_num], paste_num)
+           : zero_anchors_;
+  }
+
+  USE_OPERATOR_MEMBERS();
+  DALIDataType output_type_arg_, output_type_, input_type_;
+
+  ArgValue<int, 1> output_size_;
+
+  ArgValue<int, 1> in_idx_;
+  ArgValue<int, 2> in_anchors_;
+  ArgValue<int, 2> in_shapes_;
+  ArgValue<int, 2> out_anchors_;
+
+  kernels::KernelManager kernel_manager_;
+
+  const int zeros_[2] = {0, 0};
+  Coords zero_anchors_;
+
+  vector<bool> no_intersections_;
+  vector<int> raw_input_size_mem_;
```

Comment:
I think it'd be safer to use int64_t anywhere representing image dimensions, since that's what TensorShape uses.

Reply:
It is set to int, as it has to be the same type as "shapes" input - the input sizes are used to create replacement tensors then. If I were to change this to int64_t, the type for "shapes" would have to become 64bit also, or both types would need to be acceptable, which would make everything much harder